src/azul/time.py

```diff
@@ -159,3 +164,37 @@ def parse_dcp2_datetime(s: str) -> datetime:
     ValueError: time data '2020-01-01T00:00:00.000000' does not match format '%Y-%m-%dT%H:%M:%S.%f%z'
     """
     return datetime.strptime(s, dcp2_datetime_format)
+
+
+def parse_datetime(s: str) -> datetime:
+    """
+    Convert an unknown formatted string into a tz-aware (UTC) datetime.
+
+    >>> parse_datetime('0001-01-01T02:03:04.000005Z')
+    datetime.datetime(1, 1, 1, 2, 3, 4, 5, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2020-01-01T02:03:04.000005+00:00')
+    datetime.datetime(2020, 1, 1, 2, 3, 4, 5, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2020-01-01T020304.000005Z')
+    datetime.datetime(2020, 1, 1, 2, 3, 4, 5, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2020-01-01T02:03:04Z')
+    datetime.datetime(2020, 1, 1, 2, 3, 4, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2020-01-01T020304Z')
+    datetime.datetime(2020, 1, 1, 2, 3, 4, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2019-12-31T19:03:04.000005-07:00')
+    datetime.datetime(2020, 1, 1, 2, 3, 4, 5, tzinfo=datetime.timezone.utc)
+
+    >>> parse_datetime('2020-01-01T02:03:04.000005')
+    Traceback (most recent call last):
+    ...
+    azul.RequirementError: ('Missing a timezone specifier', '2020-01-01T02:03:04.000005')
+    """
+    date = parser.parse(s)
```

Comment:
Don't feel comfortable relying on heuristics to detect the format. Those tend to silently fail and return garbage which goes against the strictness principle that we follow. 

The bundle version in DCP/2 is standardized so I'm guessing the diversity in formats stems from the canned bundles. Let's discuss alternatives in PL.